### PR TITLE
fix condition

### PR DESCRIPTION
**Pull request by leqiao-1**
check if user_module_loader.user_script is None

### CODE REVIEW

File: olive/passes/olive_pass.py [hunk 1]
```diff
@@ -158,7 +158,9 @@ def _validate_user_script(self, config: Dict[str, Any], user_module_loader: User
         default_config = self.default_config()
         for key, value in config.items():
             if default_config[key].is_object and isinstance(value, str):
-                assert user_module_loader is not None, f"'user_script' must be specified if a {key} is a string."
+                assert (
+                    user_module_loader is not None and user_module_loader.user_script
```
[the remark this answers is not in the document]
Comment (leqiao-1) [reply olive/passes/olive_pass.py]: Removed this check

**Review by xiaoyu-work — DISMISSED** [review]

File: olive/passes/olive_pass.py [hunk 2]
```diff
@@ -158,7 +158,7 @@ def _validate_user_script(self, config: Dict[str, Any], user_module_loader: User
         default_config = self.default_config()
         for key, value in config.items():
             if default_config[key].is_object and isinstance(value, str):
-                assert user_module_loader is not None, f"'user_script' must be specified if a {key} is a string."
+                assert user_module_loader.user_script, f"'user_script' must be specified if a {key} is a string."
```
Comment (guotuofeng) [inline olive/passes/olive_pass.py:161]: add unit test to cover this case?

File: olive/passes/olive_pass.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (trajepl) [reply olive/passes/olive_pass.py]: Could you tell which kinda bug is this? When we will trigger the bug?

[the remark this answers is not in the document]
Comment (leqiao-1) [reply olive/passes/olive_pass.py]: This assertion wants to check if user_script exists when config is provided through string, for example dataloader_func
Before this fix we only check if user_module_loader is not None, which will never fails.

**Review by trajepl — APPROVED** [review]

**Pull request merged**